pkg/analytics/tracking_utils.go

```diff
@@ -5,30 +5,34 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"go.uber.org/zap"
 	"math"
 	"net/http"
 	"time"
 
 	"github.com/klothoplatform/klotho/pkg/core"
 )
 
-var kloServerUrl = "http://srv.klo.dev"
+const kloServerUrl = "http://srv.klo.dev"
 
 type AnalyticsFile struct {
 	Id string
 }
 
-func SendTrackingToServer(bundle *Client) error {
-	postBody, _ := json.Marshal(bundle)
+func (t *Client) Send(payload *Payload) {
+	postBody, _ := json.Marshal(payload)
 	data := bytes.NewBuffer(postBody)
-	resp, err := http.Post(fmt.Sprintf("%v/analytics/track", kloServerUrl), "application/json", data)
-	if err != nil {
-		return err
+	url := t.serverUrlOverride
+	if url == "" {
+		url = kloServerUrl
 	}
+	resp, err := http.Post(fmt.Sprintf("%v/analytics/track", url), "application/json", data)
 
-	defer resp.Body.Close()
-
-	return nil
+	if err != nil {
+		zap.L().Debug(fmt.Sprintf("Failed to send metrics info. %v", err))
```

Comment:
This could cause infinite loop if the server is down. Also, the previous version returning the error to let the caller decide what to do makes the most sense IMO.

Reply:
The infinite loop was there before, too :-\ I noticed it but didn't feel like thinking about how to clean it up, heh.

Regarding returning the error: in the previous version, there was only one caller,`(*Client).track`:

```
func (t *Client) track(event string) {
	t.Event = event
	err := SendTrackingToServer(t)

	if err != nil {
		zap.L().Debug(fmt.Sprintf("Failed to send metrics info. %v", err))
	}
}
```

So:
- `SendTrackingToServer` returns a possible error
- `Track` is the _only_ invoker of `SendTrackingToServer`, and the only real value it provides is to turn that error into a log line

It felt like inlining those made sense. That said, maybe it would be clearer if I (a) un-exported `SendTrackingToServer` and (b) moved it into `client.go`?